### PR TITLE
[3.6] Add digit separators to Language Tour

### LINKED ISSUE #6144: Document support for digit separators
Implementation issue: https://github.com/dart-lang/sdk/issues/56188
Feature specification: https://github.com/dart-lang/language/blob/main/accepted/future-releases/digit-separators/feature-specification.md

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Do not merge until release -- fixes #6144

### CODE REVIEW

**Comment by MaryaBelanger** [comment]
I added `// ignore_for_file: experiment_not_enabled` to pass CI for now, but I guess that didn't do anything. @parlough, when you get a chance, can you remind me the best practice for writing code excerpts for not-yet-released features? (offline works too) Thanks!

**Comment by domesticmouse** [comment]
Personally I prefer to have PRs failing when they aren't targeting current stable. Makes it obvious that they aren't ready to land

File: src/content/language/built-in-types.md [hunk 1]
```diff
@@ -173,6 +173,18 @@ const msUntilRetry = secondsUntilRetry * msPerSecond;
...
+Multiple digit separators allow for higher level grouping.
+
+<?code-excerpt "misc/lib/language_tour/built_in_types.dart (digit-separators)"?>
+```dart
+var n1 = 1_000_000;
+var n2 = 0.000_000_000_01;
+var n3 = 0x00_14_22_01_23_45;  // MAC address
+var n4 = 555_123_4567;  // US Phone number
+var n5 = 100__000_000__000_000;  // one hundred million million!
+```
```
Comment (parlough) [inline src/content/language/built-in-types.md:187]: Consider adding a version note about the required language version.

**Comment by parlough** [comment]
> I added // ignore_for_file: experiment_not_enabled to pass CI for now, but I guess that didn't do anything. @parlough, when you get a chance, can you remind me the best practice for writing code excerpts for not-yet-released features? (offline works too) Thanks!

There are a few options:

1. If the current stable version supports the experiment, you can [enable the experiment](/tools/experiment-flags) in that project's analysis config.
2. If not or the experiment isn't fully functional and you want to land this, not adding a code excerpt for now is fine (but consider creating a TODO or an issue).
3. If you don't want this to land until the feature is stable, just let the build fail. As Brett mentions that's a good indicator of when support has landed in stable.

In this case, I think this is fine to land now if there's a version note, so the easiest solution might be to just drop the excerpt.

File: src/content/language/built-in-types.md [hunk 2]
```diff
@@ -159,7 +159,7 @@ assert((3 & 4) == 0); // 0011 & 0100 == 0000
 For more examples, see the
 [bitwise and shift operator][] section.
 
-Literal numbers are compile-time constants.
+Number literals are compile-time constants.
```
Comment (srawlins) [inline src/content/language/built-in-types.md:162]: 👍

File: src/content/language/built-in-types.md (hunk 1 above)
Comment (srawlins) [reply src/content/language/built-in-types.md:187]: Ah yeah this only works if your pubspec specifies `sdk: ^3.6.0` or similar.

**Review by srawlins — APPROVED** [review]
Excellent love this, thanks!

**Review by parlough — APPROVED** [review]
Nice! Thanks for this @MaryaBelanger 🚀

**Pull request merged**